devito/dse/backends/advanced.py

```diff
@@ -7,7 +7,8 @@
 from devito.symbolics import Eq, estimate_cost, xreplace_constrained, iq_timeinvariant
 from devito.dse.manipulation import (common_subexprs_elimination, collect_nested,
                                      compact_temporaries)
-from devito.types import Indexed, Scalar, Array
+from devito.types import Indexed
+from devito.types.basic import Array, Scalar
```

Comment:
why split over two imports?

OK, this is the last one I comment -- check all imports, it feels like some are uselessly split over multiple lines